### PR TITLE
Bump OTel core components to stable 1.2.0 release

### LINKED ISSUE #468: Bump OTel .NET references to 1.2.0
I know that you are documenting current state. April 15 is expected official release for 1.2.0.
I think that it will be good idea to bump it before our beta release.

See https://docs.google.com/document/d/1yjjD6aBcLxlRazYrawukDgrhZMObwHARJbB9glWdHj8/edit#

_Originally posted by @Kielek in https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/pull/467#discussion_r847225050_

### PR DESCRIPTION
**Pull request by nrcventura**
Fixes #468 

Changes proposed in this pull request:
- Updates all of the core component dependencies from 1.2.0-rc5 to 1.2.0
- Updates the docs to reflect the dependency on the 1.2.0 core components release

### CODE REVIEW

File: test/test-applications/integrations/aspnet/TestApplication.AspNet/Web.config [hunk 1]
```diff
@@ -68,6 +68,10 @@
         <assemblyIdentity name="System.Web.Mvc" publicKeyToken="31bf3856ad364e35" />
         <bindingRedirect oldVersion="1.0.0.0-5.2.7.0" newVersion="5.2.7.0" />
       </dependentAssembly>
+      <dependentAssembly>
+        <assemblyIdentity name="System.Numerics.Vectors" publicKeyToken="b03f5f7f11d50a3a" culture="neutral" />
```
Comment (pjanotti) [inline test/test-applications/integrations/aspnet/TestApplication.AspNet/Web.config:72]: Is this a new dependency added to 1.2.0?

File: CHANGELOG.md [hunk 2]
```diff
@@ -10,7 +10,7 @@ This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.htm
 The is an initial, official beta release,
 built on top of [OpenTelemetry .NET](https://github.com/open-telemetry/opentelemetry-dotnet):
 
-- Core components: [`1.2.0-rc5`](https://github.com/open-telemetry/opentelemetry-dotnet/releases/tag/core-1.2.0-rc5)
+- Core components: [`1.2.0`](https://github.com/open-telemetry/opentelemetry-dotnet/releases/tag/core-1.2.0)
```
Comment (pjanotti) [inline CHANGELOG.md:13]:
```suggestion
- [Core components](https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/VERSIONING.md#core-components): [`1.2.0`](https://github.com/open-telemetry/opentelemetry-dotnet/releases/tag/core-1.2.0)
```

**Review by rajkumar-rangaraj — APPROVED** [review]

File: docs/README.md [hunk 3]
```diff
@@ -7,7 +7,7 @@ to .NET applications without having to modify their source code.
 
 OpenTelemetry .NET Automatic Instrumentation is built on top of [OpenTelemetry .NET](https://github.com/open-telemetry/opentelemetry-dotnet):
 
-- Core components: [`1.2.0-rc5`](https://github.com/open-telemetry/opentelemetry-dotnet/releases/tag/core-1.2.0-rc5)
+- Core components: [`1.2.0`](https://github.com/open-telemetry/opentelemetry-dotnet/releases/tag/core-1.2.0)
```
Comment (pjanotti) [inline docs/README.md:10]:
```suggestion
- [Core components](https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/VERSIONING.md#core-components): [`1.2.0`](https://github.com/open-telemetry/opentelemetry-dotnet/releases/tag/core-1.2.0)
```

File: test/test-applications/integrations/aspnet/TestApplication.AspNet/TestApplication.AspNet.csproj [hunk 4]
```diff
@@ -1,4 +1,4 @@
-<?xml version="1.0" encoding="utf-8"?>
+﻿<?xml version="1.0" encoding="utf-8"?>
```
Comment (pjanotti) [inline test/test-applications/integrations/aspnet/TestApplication.AspNet/TestApplication.AspNet.csproj:1]: Just being extra careful here: what is the whitespace change here? Notice also the last line.

**Review by pjanotti — COMMENTED** [review]
LGTM but a few Qs and a suggestion regarding "Core components"

File: test/test-applications/integrations/aspnet/TestApplication.AspNet/Web.config (hunk 1 above)
Comment (nrcventura) [reply test/test-applications/integrations/aspnet/TestApplication.AspNet/Web.config:72]: If it's a new dependency, it's a very indirect one. It was added when I triggered the update for the OTel dependencies via the Nuget package manager UI. Unfortunately I could not run that test locally to confirm whether or not it was really necessary.

File: test/test-applications/integrations/aspnet/TestApplication.AspNet/TestApplication.AspNet.csproj (hunk 4 above)
Comment (nrcventura) [reply test/test-applications/integrations/aspnet/TestApplication.AspNet/TestApplication.AspNet.csproj:1]: I think that it has to do with the byte-order mark that Visual Studio manages.

File: test/test-applications/integrations/aspnet/TestApplication.AspNet/Web.config (hunk 1 above)
Comment (nrcventura) [reply test/test-applications/integrations/aspnet/TestApplication.AspNet/Web.config:72]: @pjanotti discovered that there was a build warning on our main branch related to the "System.Numerics.Vectors" binding redirect. The Nuget package management tool in the Visual Studio UI resolved that warning for us automatically by adding this binding redirect. It is not strictly necessary for the purposes of this PR, but it is still useful.

**Review by pjanotti — APPROVED** [review]

**Review by Kielek — APPROVED** [review]

**Review by pellared — APPROVED** [review]

**Comment by pellared** [comment]
```
Error: [xUnit.net 00:00:23.46]     IntegrationTests.MongoDB.MongoDbTests.SubmitsTraces [FAIL]
  07:29:54 [ERR] [xUnit.net 00:00:23.46]     IntegrationTests.MongoDB.MongoDbTests.SubmitsTraces [FAIL]
  07:29:54 [DBG]   Failed IntegrationTests.MongoDB.MongoDbTests.SubmitsTraces [1 ms]
  07:29:54 [DBG]   Error Message:
  07:29:54 [DBG]    System.AggregateException : One or more errors occurred. (One or more errors occurred. (The operation has timed out.)) (The following constructor parameters did not have matching fixture data: MongoDbFixture mongoDb)
  07:29:54 [DBG] ---- System.AggregateException : One or more errors occurred. (The operation has timed out.)
  07:29:54 [DBG] -------- System.TimeoutException : The operation has timed out.
  07:29:54 [DBG] ---- The following constructor parameters did not have matching fixture data: MongoDbFixture mongoDb
  07:29:54 [DBG]   Stack Trace:
  07:29:54 [DBG]   
  07:29:54 [DBG] ----- Inner Stack Trace #1 (System.AggregateException) -----
  07:29:54 [DBG]    at System.Threading.Tasks.Task.ThrowIfExceptional(Boolean includeTaskCanceledExceptions)
  07:29:54 [DBG]    at System.Threading.Tasks.Task.Wait(Int32 millisecondsTimeout, CancellationToken cancellationToken)
  07:29:54 [DBG]    at System.Threading.Tasks.Task.Wait()
  07:29:54 [DBG]    at IntegrationTests.MongoDB.MongoDbFixture.LaunchMongoContainer(Int32 port) in D:\a\opentelemetry-dotnet-instrumentation\opentelemetry-dotnet-instrumentation\test\integration-tests\IntegrationTests.MongoDB\MongoDbCollection.cs:line 96
  07:29:54 [DBG]    at IntegrationTests.MongoDB.MongoDbFixture..ctor() in D:\a\opentelemetry-dotnet-instrumentation\opentelemetry-dotnet-instrumentation\test\integration-tests\IntegrationTests.MongoDB\MongoDbCollection.cs:line 49
  07:29:54 [DBG] ----- Inner Stack Trace -----
  07:29:54 [DBG]    at System.IO.Pipes.NamedPipeClientStream.ConnectInternal(Int32 timeout, CancellationToken cancellationToken, Int32 startTime)
  07:29:54 [DBG]    at System.IO.Pipes.NamedPipeClientStream.<>c__DisplayClass20_0.<ConnectAsync>b__0()
  07:29:54 [DBG]    at System.Threading.Tasks.Task.InnerInvoke()
  07:29:54 [DBG]    at System.Threading.Tasks.Task.<>c.<.cctor>b__274_0(Object obj)
  07:29:54 [DBG]    at System.Threading.ExecutionContext.RunFromThreadPoolDispatchLoop(Thread threadPoolThread, ExecutionContext executionContext, ContextCallback callback, Object state)
  07:29:54 [DBG] --- End of stack trace from previous location where exception was thrown ---
  07:29:54 [DBG]    at System.Threading.Tasks.Task.ExecuteWithThreadLocal(Task& currentTaskSlot, Thread threadPoolThread)
  07:29:54 [DBG] --- End of stack trace from previous location where exception was thrown ---
  07:29:54 [DBG]    at Docker.DotNet.DockerClient.<>c__DisplayClass6_0.<<-ctor>b__0>d.MoveNext()
  07:29:54 [DBG] --- End of stack trace from previous location where exception was thrown ---
  07:29:54 [DBG]    at Microsoft.Net.Http.Client.ManagedHandler.ProcessRequestAsync(HttpRequestMessage request, CancellationToken cancellationToken)
  07:29:54 [DBG]    at Microsoft.Net.Http.Client.ManagedHandler.SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
  07:29:54 [DBG]    at System.Net.Http.HttpClient.FinishSendAsyncBuffered(Task`1 sendTask, HttpRequestMessage request, CancellationTokenSource cts, Boolean disposeCts)
  07:29:54 [DBG]    at Docker.DotNet.DockerClient.PrivateMakeRequestAsync(TimeSpan timeout, HttpCompletionOption completionOption, HttpMethod method, String path, IQueryString queryString, IDictionary`2 headers, IRequestContent data, CancellationToken cancellationToken)
  07:29:54 [DBG]    at Docker.DotNet.DockerClient.PrivateMakeRequestAsync(TimeSpan timeout, HttpCompletionOption completionOption, HttpMethod method, String path, IQueryString queryString, IDictionary`2 headers, IRequestContent data, CancellationToken cancellationToken)
  07:29:54 [DBG]    at Docker.DotNet.DockerClient.MakeRequestAsync(IEnumerable`1 errorHandlers, HttpMethod method, String path, IQueryString queryString, IRequestContent body, IDictionary`2 headers, TimeSpan timeout, CancellationToken token)
  07:29:54 [DBG]    at Docker.DotNet.ExecOperations.InspectContainerExecAsync(String id, CancellationToken cancellationToken)
  07:29:54 [DBG]    at DotNet.Testcontainers.Clients.DockerContainerOperations.ExecAsync(String id, IList`1 command, CancellationToken ct)
  07:29:54 [DBG]    at DotNet.Testcontainers.Containers.WaitStrategies.Unix.UntilCommandIsCompleted.Until(Uri endpoint, String id)
  07:29:54 [DBG]    at DotNet.Testcontainers.Containers.WaitStrategies.WaitStrategy.<>c__DisplayClass1_0.<<WaitUntil>b__0>d.MoveNext()
  07:29:54 [DBG] --- End of stack trace from previous location where exception was thrown ---
  07:29:54 [DBG]    at DotNet.Testcontainers.Containers.WaitStrategies.WaitStrategy.WaitUntil(Func`1 wait, Int32 frequency, Int32 timeout, CancellationToken ct)
  07:29:54 [DBG]    at DotNet.Testcontainers.Containers.Modules.TestcontainersContainer.Start(String id, CancellationToken ct)
  07:29:54 [DBG]    at DotNet.Testcontainers.Containers.Modules.TestcontainersContainer.StartAsync(CancellationToken ct)
  07:29:54 [DBG] ----- Inner Stack Trace #2 (Xunit.Sdk.TestClassException) -----
  07:29:54 [DBG] 
  07:29:54 [DBG] Results File: D:\a\opentelemetry-dotnet-instrumentation\opentelemetry-dotnet-instrumentation\build_data\results\IntegrationTests.MongoDB\runneradmin_fv-az449-659_2022-04-19_07_29_15.trx
  07:29:54 [DBG] 
  07:29:54 [DBG] Failed!  - Failed:     1, Passed:     0, Skipped:     0, Total:     1, Duration: < 1 ms - IntegrationTests.MongoDB.dll (netcoreapp3.1)
```

**Pull request merged**